Update of the DT migration to DD4HEP

slomeo opened the pull request:
**PR description:**

Update the DT migration to DD4hep after @bsunanda PRs #29842 #29943 and @ianna PR #29954.

**PR validation:**

1) validation made by Geometry/DTGeometryBuilder/test/python/validateDTGeometry_cfg.py

2) validation by runTheMatrix.py -l 25202.1 :

25202.1_TTbar_13+TTbar_13+DIGIUP15APVSimu_PU25+RECOUP15_PU25+HARVESTUP15_PU25 Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED  - time date Tue Aug  4 11:08:53 2020-date Tue Aug  4 10:46:39 2020; exit: 0 0 0 0 1 1 1 1 tests passed, 0 0 0 0 failed

**if this PR is a backport please specify the original PR and why you need to backport that PR:**

nothing special

slomeo committed:
Update of the DT migration to DD4hep (step1)

slomeo committed:
update before rebase

slomeo committed:
made rebase

slomeo committed:
validation script fails, but compilation and matrix test are fine

slomeo committed:
deleted comments and unsed .h .cc files

slomeo committed:
Validation Script ok tks to Carl's fix related to DTNumberingScheme.cc

slomeo committed:
Ready for PR

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-31043/17542

- This PR adds an extra 28KB to repository

slomeo committed:
Date fixed

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-31043/17545

- This PR adds an extra 28KB to repository

cmsbuild commented:
Pull request #31043 was updated. @civanch, @Dr15Jones, @makortel, @cvuosalo, @ianna, @mdhildreth, @cmsbuild can you please check and sign again.

cvuosalo commented on Geometry/MuonNumbering/src/DTNumberingScheme.cc:
```diff
@@ -32,7 +42,8 @@ int DTNumberingScheme::baseNumberToUnitNumber(const MuonBaseNumber& num) const {
 #endif
-  if (num.getLevels() != theWireLevel) {
+  //  if (num.getLevels() != theWireLevel) {
+  if (num.getLevels() < theStationLevel) {
```
Are you sure you want to keep this version of the `if` statement? What is the correct number of levels? Have you talked to Sunanda about this? He recently fixed a similar problem with the GEMs.
Whatever you decide, make sure the message on line 48 matches the `if`.

slomeo replied:
@cvuosalo I didn't talk with @bsunanda 

>>He recently fixed a similar problem with the GEMs.

@bsunanda how can you fix it for GEMs?

slomeo replied:
@cvuosalo I fixed line 48

slomeo replied:
@cvuosalo probably it could be better to go forward with this PR in order to test it (i.e "please test") in order to see if all is fine, then I will get in touch (offline) with @bsunanda and you in order to understand better the number levels issue

bsunanda replied:
@slomeo I just give a warning sign and proceed. I test the number of levels at each step before assigning chamber, roll, ...

slomeo committed:
fixed line 48 in DTNumberingScheme.cc

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-31043/17554

- This PR adds an extra 28KB to repository

cmsbuild commented:
Pull request #31043 was updated. @civanch, @Dr15Jones, @makortel, @cvuosalo, @ianna, @mdhildreth, @cmsbuild can you please check and sign again.

cvuosalo commented on Geometry/MuonNumbering/src/DTNumberingScheme.cc:
```diff
@@ -32,9 +42,9 @@ int DTNumberingScheme::baseNumberToUnitNumber(const MuonBaseNumber& num) const {
+  if (num.getLevels() < theStationLevel) {  // it was  if (num.getLevels() != theWireLevel) {
     edm::LogWarning("DTNumberingScheme") << "DTNumberingScheme::BNToUN: BaseNumber has " << num.getLevels()
-                                         << " levels, need " << theWireLevel;
+                                         << " levels, need " << theWireLevel;  //it was theWireLevel;
```
```suggestion
                                         << " levels, need " << theStationLevel;  //it was theWireLevel;
```

slomeo replied:
@cvuosalo  : done

cvuosalo commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1

Tested at: a64cab2da1d6e1a718ba2eee2b3752dfcabdebd1

CMSSW: CMSSW_11_2_X_2020-08-04-1100
SCRAM_ARCH: slc7_amd64_gcc820
You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c42353/8569/summary.html

I found follow errors while testing this PR

**Failed tests:  UnitTests**

* **Unit Tests**:

I found errors in the following unit tests: 
 
 ---> test GeometryDTGeometryBuilderTestDriver had ERRORS

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c42353/8569/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 4 differences found in the comparisons
 - DQMHistoTests: Total files compared: 34
 - DQMHistoTests: Total histograms compared: 2525448
 - DQMHistoTests: Total failures: 7
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2525394
 - DQMHistoTests: Total skipped: 47
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 33 files compared)
 - Checked 144 log files, 17 edm output root files, 34 DQM output files

slomeo commented:
@cvuosalo @ianna @bsunanda I took a look to the log file of the test (i.e Unit Tests) and I saw that the validateDTGeometry_cfg.py fails because: "ERROR: failed to find geometry file. Initialization failed." I have no problem with this script because in my local directory CMSSW_11_2_X_2020-08-03-2300/src/ it's present the Geometry file (i.e cmsRecoGeom-2021.root). The validate script is called by CMSSW_11_2_X_2020-08-04-1100/src/Geometry/DTGeometryBuilder/test/runTest.sh and it tried to find the Geometry file in CMSSW_11_2_X_2020-08-04-1100 (and not in CMSSW_11_2_X_2020-08-03-2300). How can I do to fix this problem?

slomeo committed:
fixed typo in DTNumberingScheme

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-31043/17560

- This PR adds an extra 28KB to repository

cmsbuild commented:
Pull request #31043 was updated. @civanch, @Dr15Jones, @makortel, @cvuosalo, @ianna, @mdhildreth, @cmsbuild can you please check and sign again.

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-31043/17561

- This PR adds an extra 52KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  Configuration/PyReleaseValidation/python/relval_2026.py  modified in PR(s): #30976
  - File  Configuration/PyReleaseValidation/python/relval_steps.py  modified in PR(s): #31020
  - File  Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py  modified in PR(s): #30976
  - File  PhysicsTools/NanoAOD/python/met_cff.py  modified in PR(s): #30922
  - File  PhysicsTools/NanoAOD/python/nanoDQM_cfi.py  modified in PR(s): #30922
  - File  PhysicsTools/NanoAOD/python/nano_cff.py  modified in PR(s): #30306, #30922
  - File  PhysicsTools/PatAlgos/python/slimming/MicroEventContent_cff.py  modified in PR(s): #30544, #30898
  - File  PhysicsTools/PatAlgos/python/slimming/slimming_cff.py  modified in PR(s): #30544

cmsbuild commented:
Pull request #31043 was updated. @gouskos, @chayanit, @wajidalikhan, @ianna, @kpedro88, @Martin-Grunewald, @jpata, @santocch, @peruzzim, @perrotta, @civanch, @makortel, @cmsbuild, @fwyzard, @davidlange6, @smuzaffar, @Dr15Jones, @cvuosalo, @mdhildreth, @slava77, @ggovi, @qliphy, @fabiocos, @pgunnell, @silviodonato, @franzoni, @fgolf, @mariadalfonso can you please check and sign again.

civanch commented:
@slomeo , are you sure all these files are needed for this PR?

slomeo commented:
@civanch : no, they aren't. After a rebase it occurs this problem... now I'll perform a squash

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-31043/17563

- This PR adds an extra 24KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  CalibTracker/SiStripCommon/interface/ShallowEventDataProducer.h  modified in PR(s): #31002
  - File  CalibTracker/SiStripCommon/interface/ShallowGainCalibration.h  modified in PR(s): #31002
  - File  CalibTracker/SiStripCommon/interface/ShallowTracksProducer.h  modified in PR(s): #31002
  - File  CalibTracker/SiStripCommon/plugins/ShallowGainCalibration.cc  modified in PR(s): #31002
  - File  Configuration/AlCa/python/autoCond.py  modified in PR(s): #31020, #31050
  - File  Configuration/Geometry/python/dict2026Geometry.py  modified in PR(s): #30976, #30126
  - File  Configuration/PyReleaseValidation/python/relval_2026.py  modified in PR(s): #30976
  - File  Configuration/PyReleaseValidation/python/relval_steps.py  modified in PR(s): #31020
  - File  Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py  modified in PR(s): #30976
  - File  Geometry/CMSCommonData/python/cmsExtendedGeometry2026D49XML_cfi.py  modified in PR(s): #30126
  - File  Geometry/CMSCommonData/python/cmsExtendedGeometry2026D50XML_cfi.py  modified in PR(s): #30126
  - File  Geometry/CMSCommonData/python/cmsExtendedGeometry2026D53XML_cfi.py  modified in PR(s): #30126
  - File  PhysicsTools/NanoAOD/python/met_cff.py  modified in PR(s): #30922
  - File  PhysicsTools/NanoAOD/python/nanoDQM_cfi.py  modified in PR(s): #30922
  - File  PhysicsTools/NanoAOD/python/nano_cff.py  modified in PR(s): #30306, #30922
  - File  PhysicsTools/PatAlgos/python/slimming/MicroEventContent_cff.py  modified in PR(s): #30544, #30898
  - File  PhysicsTools/PatAlgos/python/slimming/slimming_cff.py  modified in PR(s): #30544


Code check has found code style and quality issues which could be resolved by applying following patch(s)
- **code-checks**:
https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-31043/17563/code-checks.patch
e.g. `curl https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-31043/17563/code-checks.patch | patch -p1`
You can also run `scram build code-checks` to apply code checks directly
- **code-format**:
https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-31043/17563/code-format.patch
e.g. `curl https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-31043/17563/code-format.patch | patch -p1`
You can also run `scram build code-format` to apply code format directly

slomeo commented:
@civanch I'll try to restart in a clean IB. I don't know what's wrong

slomeo committed:
restart

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-31043/17565

- This PR adds an extra 32KB to repository

cmsbuild commented:
Pull request #31043 was updated. @civanch, @Dr15Jones, @makortel, @cvuosalo, @ianna, @mdhildreth, @cmsbuild can you please check and sign again.

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-31043/17574

- This PR adds an extra 32KB to repository

cmsbuild commented:
Pull request #31043 was updated. @civanch, @Dr15Jones, @makortel, @cvuosalo, @ianna, @mdhildreth, @cmsbuild can you please check and sign again.

slomeo commented:
@cvuosalo @civanch I made a rebase in the last IB, please can you request a "test"?

cvuosalo commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1

Tested at: 5548d90182ec5cf1d75a5ed2ed60192524de272b

CMSSW: CMSSW_11_2_X_2020-08-04-2300
SCRAM_ARCH: slc7_amd64_gcc820
You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c42353/8586/summary.html

I found follow errors while testing this PR

**Failed tests:  UnitTests**

* **Unit Tests**:

I found errors in the following unit tests: 
 
 ---> test GeometryDTGeometryBuilderTestDriver had ERRORS

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c42353/8586/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 35
 - DQMHistoTests: Total histograms compared: 2612401
 - DQMHistoTests: Total failures: 1
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2612352
 - DQMHistoTests: Total skipped: 48
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 34 files compared)
 - Checked 149 log files, 22 edm output root files, 35 DQM output files

slomeo commented:
@cvuosalo I made a test:

I edited the file /build/slomeo/CMSSW_11_2_X_2020-08-04-2300/src/Geometry/DTGeometryBuilder/test/runTest.sh (it fails during the "cmsbuild test") in this way:

#F1=${LOCAL_TEST_DIR}/python/validateDTGeometry_cfg.py                                                                                                                                            
#F2=${LOCAL_TEST_DIR}/python/testDTGeometry.py                                                                                                                                                    
#FILE1=${LOCAL_TEST_DIR}/dtGeometry.log.org                                                                                                                                                       
F1=/build/slomeo/CMSSW_11_2_X_2020-08-04-2300/src/Geometry/DTGeometryBuilder/test/python/validateDTGeometry_cfg.py
F2=/build/slomeo/CMSSW_11_2_X_2020-08-04-2300/src/Geometry/DTGeometryBuilder/test/python/testDTGeometry.py
FILE1=/build/slomeo/CMSSW_11_2_X_2020-08-04-2300/src/Geometry/DTGeometryBuilder/test/dtGeometry.log.org

and I copied the cmsRecoGeom-2021.root file (i.e the Geometry Reco file) inside /build/slomeo/CMSSW_11_2_X_2020-08-04-2300/src/Geometry/DTGeometryBuilder/test/

and now runTest.sh works, but IMHO the latter is not necessary anymore. What do you think about this? And @ianna ?

Indeed, the output (OK) of the validateDTGeometry_cfg.py script is the "conditio sine qua non" (for me) in order to submit a PR (as I always did for every Muon Subsystem), so it is not necessary to re-run the script.

But If you want to leave runTest.sh I suggest to re-write it leaving only "cmsRun validateDTGeometry_cfg.py" but in this case it is needed:
1) to have the reco geometry file present inside the test directory
2) set the name of the test directory (i.e I don't think that /build/slomeo ... is right...)
so it is necessary to write bash commands to perform these job.

There is a second point: taking a look to https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c42353/8586/summary.html you can see that there is a problem in "Static Check Build Log". 

There are some warning and one error:
---
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_2_X_2020-08-04-2300/src/Geometry/MuonNumbering/src/GEMNumberingScheme.cc:38:67: error: use of undeclared identifier 'maxLevel'
                                           << " levels, need " << maxLevel << ":" << theRingLevel; 
---
It is related to GEMNumberingScheme.cc, please @bsunanda can you check?

In conclusion, please @cvuosalo @ianna @bsunanda let me know

cvuosalo commented:
@slomeo I think the unit test needs to be kept. When you change the input file, you have to add the new input file to the external data file repository.
@smuzaffar Could you please help Sergio add a file to the external data file repository?

slomeo commented:
@cvuosalo ok, but IMHO I'd leave, in runTest.sh, only the call to validateDTGeometry_cfg.py (the script I used for all the migrations) and I'd delete all the other lines. Do you agree?

cvuosalo commented:
@slomeo Why do you want to eliminate the `testDTGeometry.py` test? Does it cause any problem?

slomeo commented:
@cvuosalo : The question is not if testDTGeometry.py works, but if it is necessary (now) for this PR; let me clarify:

When the migration (for Muon) started my "modus operandi" was:
1) cross-check DDD Geometry vs RECO Geometry (i.e create a validationDD.root that contains some comparison histos)
2) cross-check DD4HEP Geometry vs RECO Geometry (i.e create a validationDD4HEP.root that contains some comparison histos)
3) If the histos present in both the files validationDD.root and validationDD4HEP.root were the same and the runTheMatrix test was ok --> PR

When all Muon Subsystems were migrated, @bsunanda has changed (a nice idea/job) the MuonNumbering so the migration of all Muon Subsystems needed an update and I started to perform this update. 

During each update I checked "only" that the comparison histos, between RECO Geometry and DD4HEP Geometry, were all centered in zero (i.e I have not checked the DDD part anymore) and 
after every update was merged, @bsunanda modified the validation script for each Muon Subsystem in order to run both DDD and DD4HEP toghether.

As far I understood, for DTs (please @ianna tell me if I'm wrong), the runTest.sh script contains the DD4HEP validation script and the DDD (testDTGeometry.py) validation. What I propose to do is  to leave, inside the runTest.sh, only the DD4HEP validation and comment the lines related to DDD. After this PR will be merged, we can let @bsunanda starts the last update of the validation script for DTs (as he did with the other Muon Subsystems).

In this way, during the @bsunanda ' update (of DTs) I can start to work on @fabiocos ' s PR https://github.com/cms-sw/cmssw/pull/30003 in order to create two MC samples for Z --> Mu Mu @ 14 TeV for DDD and DD4HEP geometries that what MUON colleagues asked in order to check if there are differences between DDD and DD4HEP (i.e checking all the histos from step4) 

Obviously what I wrote above are just my opinions to speed things up.

Please let me know.

cvuosalo commented:
@slomeo `testDTGeometry.py` is a simple sanity check that the geometry can be built from XML with DD4hep, as far as I can tell. It is not a strong test, nor a sufficient one, but it does test basic functionality. While we are still in the process of migration, I don't see the need to remove tests. I think we should keep it unless there is a strong argument to remove it. Is it too slow or taking too long?

slomeo commented:
@cvuosalo ok, but if you take a look to testDTGeometry.py you can find that the lines from 44 to 66 are the same for validateDTGeometry_cfg.py (from line 10 to line 32) so I think that these two tests are quite similar. Now I'll perform a new commit

slomeo committed:
changed runTest.sh

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-31043/17695

- This PR adds an extra 52KB to repository

cmsbuild commented:
Pull request #31043 was updated. @civanch, @Dr15Jones, @makortel, @cvuosalo, @ianna, @mdhildreth, @cmsbuild can you please check and sign again.

slomeo commented:
@cvuosalo : as I said in a previous post, in this new "commit" I added, in runTest.sh, the complete path for each test (i.e testDTgeometry.py and validateDTGeometry_cfg.py) and I added the Reco .root file inside the test directory. If the "cmsbuild test" will be ok (I don't think it will be ok) we need to change the lines in order to have a general path (i.e not ../slomeo/...)

cvuosalo commented:
@slomeo I don't understand why the `build/slomeo` directory is used in the `runTest.sh`. Why don't the commented-out lines work?

slomeo commented:
@cvuosalo : I don't know. It seems that the variable LOCAL_TEST_DIR is empty. 

If I set again:
F1=${LOCAL_TEST_DIR}/python/validateDTGeometry_cfg.py
F2=${LOCAL_TEST_DIR}/python/testDTGeometry.py
FILE1=${LOCAL_TEST_DIR}/dtGeometry.log.org
I obtained:

[slomeo@cmsdev20 test]$ ./runTest.sh 
 testing Geometry/DTGeometryBuilder
===== Test "cmsRun validateDTGeometry_cfg.py" ====
----- Begin Fatal Exception 11-Aug-2020 18:49:08 CEST-----------------------
An exception of category 'StdException' occurred while
   [0] Processing the python configuration file named /python/validateDTGeometry_cfg.py
Exception Message:
A std::exception was thrown.
File "/python/validateDTGeometry_cfg.py" could not be opened!
----- End Fatal Exception -------------------------------------------------
Failure using cmsRun /python/validateDTGeometry_cfg.py: status 66

cvuosalo commented:
@slomeo The testing environment has to be set up. When `scram b runtests` runs, it sets up the proper environment. If you run the tests individually at the command line, you are responsible for setting up the environment.

slomeo commented:
@cvuosalo : ok, now I understand. I will commit the changes asap. But what about the external source (i.e Reco Geometry .root file)?

cvuosalo commented:
@smuzaffar @mrodozov Could you please help @slomeo add a new file to the external data file repository?
If Shahzad is on vacation, maybe @kpedro88 knows how to add to the external data file repository?

kpedro88 commented:
@cvuosalo does the external data repository already exist?
if so:
1. make a fork
2. clone the fork locally
3. add the file, commit the change, push
4. send a PR from fork to upstream repo

mrodozov commented:
@kpedro88 I think we'll need  Geometry-DTGeometryBuilder repo name in cms-data
there is none with such name
https://github.com/cms-sw/cmssw/pull/31043/files#diff-6982e6763ecb0c0584fb7e49cc84ad6cR85
but let @slomeo confirms first this is the right category - Geometry-DTGeometryBuilder
@slomeo when the repo is created, this is how a data file is referenced:
https://github.com/cms-sw/cmssw/pull/31036/files#diff-969b508e7b9a9e2db5f0f105ca02046aR15
and here is how file(s) is(are) added:
https://github.com/cms-data/RecoBTag-Combined/pull/34/files

slomeo commented:
@mrodozov Yes, it is the right category but cmsRecoGeom-2021.root is a Reco file for all 2021 scenario, so maybe is it better to have a more general category (not only for Dts): please @cvuosalo can you comment?

@mrodozov I don't understand exactly how to add a .root file because in https://github.com/cms-sw/cmssw/pull/31036/files#diff-969b508e7b9a9e2db5f0f105ca02046aR15 and https://github.com/cms-data/RecoBTag-Combined/pull/34/files I didn't see any examples/refs of .root file added, can you please point me what lines I have to modify/add?

mrodozov commented:
@slomeo yes , but the file with the .onnx extension, it's a data file as well, a binary (same as if it was .root)

For example this two lines are referring to a .json and a .onnx files:
https://github.com/cms-sw/cmssw/pull/31036/files#diff-969b508e7b9a9e2db5f0f105ca02046aR23-R24

You'll have to refer to your .root file in this constructor:
https://github.com/cms-sw/cmssw/pull/31043/files#diff-6982e6763ecb0c0584fb7e49cc84ad6cR83-R85
like this:

```
DTGeometryValidate::DTGeometryValidate(const edm::ParameterSet& iConfig)
    : dtGeometryToken_{esConsumes<DTGeometry, MuonGeometryRecord>(edm::ESInputTag{})},
      infileName_(iConfig.getUntrackedParameter<string>("infileName", 
"Geometry/DTGeometryBuilder/data/cmsRecoGeom-2021.root")),
      outfileName_(iConfig.getUntrackedParameter<string>("outfileName", 
"Geometry/DTGeometryBuilder/data/validateDTGeometry.root")),
      tolerance_(iConfig.getUntrackedParameter<int>("tolerance", 6)) {
  fwGeometry_.loadMap(infileName_.c_str());
  outFile_ = TFile::Open(outfileName_.c_str(), "RECREATE");
}
```

slomeo commented:
@mrodozov tks! So, in my constructor instead of: "infileName_(iConfig.getUntrackedParameter<string>("infileName", 
"Geometry/DTGeometryBuilder/data/cmsRecoGeom-2021.root"))" I have to write: "infileName_(iConfig.getUntrackedParameter<string>("infileName", 
"PATH_OF_THE_REPOSITORY_CALLED_DTGEOMETRYBUILDER/cmsRecoGeom-2021.root"))". Am I right?

mrodozov commented:
Right, it's a string reference, "package/subpackage/data/filename.extension".
and "Geometry/DTGeometryBuilder/data/cmsRecoGeom-2021.root" will be exactly as it is

cvuosalo commented:
@slomeo Yes, a general Geometry category for the data files would seem to be adequate.

slomeo commented:
@cvuosalo : ok, will be @mrodozov and/or @kpedro88 to create this Geometry data repository?

civanch commented:
Hi all,

I would think root file should be external and not a part of  any X/Y/data directory. The example: HF and Castor shower libraries.

civanch commented:
@slomeo , all, To say more: ideally git control text files. Binary files should not be under git supervision.

kpedro88 commented:
@civanch the proposal is to use a separate repository in the cms-data organization to host the root files as externals

civanch commented:
@kpedro88 , I hope it is not git repository. We have many external files as HF SL - they are not in git.

kpedro88 commented:
@civanch as you can see from https://github.com/cms-data, many CMSSW packages already use this approach. Because these repositories only host data, and not code, nobody's workflow is impacted by the inefficiency of storing binary files in git.

mrodozov commented:
> @kpedro88 , I hope it is not git repository. We have many external files as HF SL - they are not in git.

there is github policy not to push executables. I don't see why not to use version control for binaries. except they have to be up to 100mb

cvuosalo commented:
@mrodozov @kpedro88 Could you please create the Geometry data repo needed for the data file in this PR?

mrodozov commented:
I don't have permissions to do it. @kpedro88 can do it

kpedro88 commented:
I don't have permission either; it might only be @smuzaffar ...

cvuosalo commented:
@makortel @smuzaffar @Dr15Jones Is Shahzad away on vacation? Is he the only one in CMS who can create a new data file repository in cms-data? We need to finish with this PR, and we need a data file added to the repo.

smuzaffar commented:
yes I am on vacation. CMSSW_LI's ( https://github.com/cms-sw/cms-bot/blob/master/categories.py#L11 ) can create data repos. Anyway I have created https://github.com/cms-data/Geometry-DTGeometryBuilder . Is there any other data repo you need?

cvuosalo commented:
@smuzaffar Thanks. We actually want a more general repo called `https://github.com/cms-data/Geometry`. Could you please create that?

kpedro88 commented:
@cvuosalo the data repos have to be named for packages, not subsystems. what about Geometry-CMSCommonData? I've been told that can coexist with the existing https://github.com/cms-sw/cmssw/tree/master/Geometry/CMSCommonData/data

cvuosalo commented:
@smuzaffar @kpedro88 Yes, thank you, `Geometry-CMSCommonData` would be fine. We just want a more general name to reflect that the data apply to several sub-detectors.

davidlange6 commented:
How large do we foresee this package getting?

On Aug 18, 2020, at 12:21 AM, Carl Vuosalo <notifications@github.com<mailto:notifications@github.com>> wrote:



@smuzaffar<https://github.com/smuzaffar> @kpedro88<https://github.com/kpedro88> Yes, thank you, Geometry-CMSCommonData would be fine. We just want a more general name to reflect that the data apply to several sub-detectors.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/cms-sw/cmssw/pull/31043#issuecomment-675144183>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ABGPFQYG6RTSZ776CRL2243SBGUO3ANCNFSM4PUGPVWA>.

slomeo commented:
> How large do we foresee this package getting?

@davidlange6 : We need that it contains (at least) the reco .root file for the Scenario2021 
(i.e. -rw-r--r--.  1 slomeo zh 10438530 11 ago 13.37 cmsRecoGeom-2021.root)

davidlange6 commented:
I didn’t really mean what is needed today or tomorrow, but rather after the changeover to dd4hep is done

On Aug 18, 2020, at 8:40 AM, Sergio Lo Meo <notifications@github.com<mailto:notifications@github.com>> wrote:



How large do we foresee this package getting?

@davidlange6<https://github.com/davidlange6> : We need that it contains (at least) the reco .root file for the Scenario2021
(i.e. -rw-r--r--. 1 slomeo zh 10438530 11 ago 13.37 cmsRecoGeom-2021.root)

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/cms-sw/cmssw/pull/31043#issuecomment-675286658>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ABGPFQY77QGVY5OMSXW6FGDSBIO6ZANCNFSM4PUGPVWA>.

mrodozov commented:
@slomeo could you share this files somewhere , afs or somewhere else, I'd like to see how big they are.
 I can also put them in https://github.com/cms-data/Geometry-DTGeometryBuilder.

slomeo commented:
@mrodozov I put the file cmsRecoGeom-2021.root in /afs/cern.ch/user/s/slomeo/public

how can I put it in https://github.com/cms-data/Geometry-DTGeometryBuilder? 

which is the exact string path I have to write in my Constructor in order to load this file? (i.e in "infileName_(iConfig.getUntrackedParameter("infileName",
"PATH_OF_THE_REPOSITORY_CALLED_DTGEOMETRYBUILDER/cmsRecoGeom-2021.root"))")

is (only) this: "Geometry/DTGeometryBuilder/data/cmsRecoGeom-2021.root". it seems too short

cvuosalo commented:
@davidlange6 For the future size of the data repo, I don't think it would exceed a few tens of megabytes. We don't have any plans of filling it with lots of data.

davidlange6 commented:
Eg, 10MB per geometry supported by cms?

On Aug 18, 2020, at 5:15 PM, Carl Vuosalo <notifications@github.com<mailto:notifications@github.com>> wrote:



@davidlange6<https://github.com/davidlange6> For the future size of the data repo, I don't think it would exceed a few tens of megabytes. We don't have any plans of filling it with lots of data.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/cms-sw/cmssw/pull/31043#issuecomment-675540769>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ABGPFQ6EA3X3CHMH32WBXGTSBKLI5ANCNFSM4PUGPVWA>.

cvuosalo commented:
@davidlange6 I am not aware of a plan to add more data files to the data repo at this point. We could potentially need a data reference file for each major run period, but that is not decided yet. We can stay within any recommended size limits.

cvuosalo commented:
@slomeo `Geometry/DTGeometryBuilder/data/cmsRecoGeom-2021.root` should be the correct way to reference the data file, but the file isn't there yet. You need to follow Kevin's instructions to add the file:
https://github.com/cms-sw/cmssw/pull/31043#issuecomment-672119406

slomeo commented:
@cvuosalo @kpedro88 : I created a fork (i.e slomeo/Geometry-DTGeometryBuilder forked from cms-data/Geometry-DTGeometryBuilder) then in lxplus (i.e /build/slomeoCMSSW_11_2_X_2020-08-11-1100/src where I created the PR of the DT migration to DD4HEP) I performed:

git clone https://github.com/slomeo/Geometry-DTGeometryBuilder.git

Cloning into 'Geometry-DTGeometryBuilder'...
remote: Enumerating objects: 3, done.
remote: Counting objects: 100% (3/3), done.
remote: Total 3 (delta 0), reused 0 (delta 0), pack-reused 0
Unpacking objects: 100% (3/3), done.

but now, how can I add the repo .root file to Geometry-DTGeometryBuilder repo?

slomeo committed:
added line to DTGeometryValidate.cc to link the script with external depo

slomeo committed:
restored original version of runTest.sh

slomeo commented:
@mrodozov I checked https://github.com/cms-data/Geometry-DTGeometryBuilder but the reco .root file is not present.. I followed all the instructions had (see comments 4 days ago). How can I fix it?

makortel commented on Geometry/DTGeometryBuilder/plugins/dd4hep/DTGeometryESProducer.cc:
```diff
@@ -137,8 +138,6 @@ std::shared_ptr<DTGeometry> DTGeometryESProducer::produce(const MuonGeometryReco
-    // m_applyAlignment is scheduled for removal.
-    // Ideal geometry obtained by using 'fake alignment' (with m_applyAlignment = true)
     edm::ESHandle<Alignments> globalPosition;
     record.getRecord<GlobalPositionRcd>().get(m_alignmentsLabel, globalPosition);
```
Sidetracking a bit, but I want to note that these calls need to be changed to use the `ESGetToken` to get the products. Getting products by labels in was disabled already some time ago in ESProducers.

cvuosalo replied:
@makortel I made an issue about this problem: #31288

mrodozov commented:
@slomeo I though I mislead you how to reference the root file, but in yesterdays meeting it was confirmed it's ok to be referenced as described.
@davidlange6 you mentioned yesterday what was missing, could you please tell @slomeo what to add ?

smuzaffar commented on Geometry/DTGeometryBuilder/plugins/DTGeometryValidate.cc:
```diff
@@ -82,11 +82,12 @@ class DTGeometryValidate : public one::EDAnalyzer<> {
     : dtGeometryToken_{esConsumes<DTGeometry, MuonGeometryRecord>(edm::ESInputTag{})},
-      infileName_(iConfig.getUntrackedParameter<string>("infileName", "cmsGeom10.root")),
+      infileName_(
+          iConfig.getUntrackedParameter<string>("infileName", "Geometry/DTGeometryBuilder/data/cmsRecoGeom-2021.root")),
```
You need to use FileInPath to get the file e.g. for python module you need to do something like
https://github.com/cms-sw/cmssw/blob/02d4198c0b6615287fd88e9a8ff650aea994412e/CalibCalorimetry/HcalPlugins/python/Hcal_FakeConditions_cff.py#L13

and from cc code you can do something like
https://github.com/cms-sw/cmssw/blob/0f58e3682e346587094bb503ef4556a4ed472d31/DetectorDescription/Core/test/DDIsValid.cppunit.cc#L172
OR
https://github.com/cms-sw/cmssw/blob/ed1b00288c95ceccee10546314d2384ee98ce063/RecoLocalMuon/GEMRecHit/plugins/GEMRecHitProducer.cc#L37

slomeo replied:
@smuzaffar : I modified the constructor of the DTGeometryValidate.cc in this way:

..
 edm::FileInPath fp("Geometry/DTGeometryBuilder/data/cmsRecoGeom-2021.root");                                                                                                                                   fwGeometry_.loadMap(fp.fullPath().c_str());

instead of 
...
 iConfig.getUntrackedParameter<string>("infileName", "Geometry/DTGeometryBuilder/data/cmsRecoGeom-2021.root"))
....
fwGeometry_.loadMap(infileName_.c_str()); 

The compilation is ok, but when I run: "cmsRun Geometry/DTGeometryBuilder/test/pyton/validateDTGeometry_cfg.py" I see:

----- Begin Fatal Exception 27-Aug-2020 15:33:18 CEST-----------------------
An exception of category 'FileInPathError' occurred while
   [0] Constructing the EventProcessor
   [1] Constructing module: class=DTGeometryValidate label='valid'
Exception Message:
edm::FileInPath unable to find file Geometry/DTGeometryBuilder/data/cmsRecoGeom-2021.root anywhere in the search path.
The search path is defined by: CMSSW_SEARCH_PATH
${CMSSW_SEARCH_PATH} is: /build/slomeo/CMSSW_11_2_X_2020-08-23-2300/poison:/build/slomeo/CMSSW_11_2_X_2020-08-23-2300/src:/build/slomeo/CMSSW_11_2_X_2020-08-23-2300/external/slc7_amd64_gcc820/data:/cvmfs/cms-ib.cern.ch/week1/slc7_amd64_gcc820/cms/cmssw-patch/CMSSW_11_2_X_2020-08-23-2300/poison:/cvmfs/cms-ib.cern.ch/week1/slc7_amd64_gcc820/cms/cmssw-patch/CMSSW_11_2_X_2020-08-23-2300/src:/cvmfs/cms-ib.cern.ch/week1/slc7_amd64_gcc820/cms/cmssw-patch/CMSSW_11_2_X_2020-08-23-2300/external/slc7_amd64_gcc820/data
Current directory is: /build/slomeo/CMSSW_11_2_X_2020-08-23-2300/src
----- End Fatal Exception -------------------------------------------------

Where is my mistake?

davidlange6 commented:
Your external is only present starting from the 08-26-2300 IB from what I seee.

On Aug 27, 2020, at 3:38 PM, Sergio Lo Meo <notifications@github.com<mailto:notifications@github.com>> wrote:



@slomeo commented on this pull request.

________________________________

In Geometry/DTGeometryBuilder/plugins/DTGeometryValidate.cc<https://github.com/cms-sw/cmssw/pull/31043#discussion_r478425813>:

> @@ -82,11 +82,12 @@ class DTGeometryValidate : public one::EDAnalyzer<> {

 DTGeometryValidate::DTGeometryValidate(const edm::ParameterSet& iConfig)
     : dtGeometryToken_{esConsumes<DTGeometry, MuonGeometryRecord>(edm::ESInputTag{})},
-      infileName_(iConfig.getUntrackedParameter<string>("infileName", "cmsGeom10.root")),
+      infileName_(
+          iConfig.getUntrackedParameter<string>("infileName", "Geometry/DTGeometryBuilder/data/cmsRecoGeom-2021.root")),


@smuzaffar<https://github.com/smuzaffar> : I modified the constructor of the DTGeometryValidate.cc<http://DTGeometryValidate.cc> in this way:

..
edm::FileInPath fp("Geometry/DTGeometryBuilder/data/cmsRecoGeom-2021.root"); fwGeometry_.loadMap(fp.fullPath().c_str());

instead of
...
iConfig.getUntrackedParameter("infileName", "Geometry/DTGeometryBuilder/data/cmsRecoGeom-2021.root"))
....
fwGeometry_.loadMap(infileName_.c_str());

The compilation is ok, but when I run: "cmsRun Geometry/DTGeometryBuilder/test/pyton/validateDTGeometry_cfg.py" I see:

----- Begin Fatal Exception 27-Aug-2020 15:33:18 CEST-----------------------
An exception of category 'FileInPathError' occurred while
[0] Constructing the EventProcessor
[1] Constructing module: class=DTGeometryValidate label='valid'
Exception Message:
edm::FileInPath unable to find file Geometry/DTGeometryBuilder/data/cmsRecoGeom-2021.root anywhere in the search path.
The search path is defined by: CMSSW_SEARCH_PATH
${CMSSW_SEARCH_PATH} is: /build/slomeo/CMSSW_11_2_X_2020-08-23-2300/poison:/build/slomeo/CMSSW_11_2_X_2020-08-23-2300/src:/build/slomeo/CMSSW_11_2_X_2020-08-23-2300/external/slc7_amd64_gcc820/data:/cvmfs/cms-ib.cern.ch/week1/slc7_amd64_gcc820/cms/cmssw-patch/CMSSW_11_2_X_2020-08-23-2300/poison:/cvmfs/cms-ib.cern.ch/week1/slc7_amd64_gcc820/cms/cmssw-patch/CMSSW_11_2_X_2020-08-23-2300/src:/cvmfs/cms-ib.cern.ch/week1/slc7_amd64_gcc820/cms/cmssw-patch/CMSSW_11_2_X_2020-08-23-2300/external/slc7_amd64_gcc820/data<http://cms-ib.cern.ch/week1/slc7_amd64_gcc820/cms/cmssw-patch/CMSSW_11_2_X_2020-08-23-2300/poison:/cvmfs/cms-ib.cern.ch/week1/slc7_amd64_gcc820/cms/cmssw-patch/CMSSW_11_2_X_2020-08-23-2300/src:/cvmfs/cms-ib.cern.ch/week1/slc7_amd64_gcc820/cms/cmssw-patch/CMSSW_11_2_X_2020-08-23-2300/external/slc7_amd64_gcc820/data>
Current directory is: /build/slomeo/CMSSW_11_2_X_2020-08-23-2300/src
----- End Fatal Exception -------------------------------------------------

Where is my mistake?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/cms-sw/cmssw/pull/31043#discussion_r478425813>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ABGPFQ63SPLOXVQ4QU4SXETSCZOXDANCNFSM4PUGPVWA>.

slomeo commented:
@davidlange6 : ok I'll rebase. @cvuosalo @bsunanda in order to solve conflicts with IB 08-26 I see that, in the Master, it has been modified the file Geometry/MuonNumbering/interface/DD4hep_DTNumberingScheme.h. I think that there will be some problems... because this file, as I did for RPC, CSC..., has to be deleted.

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
-code-checks

ERROR: Build errors found during clang-tidy run.
```
Geometry/DTGeometryBuilder/plugins/dd4hep/DTGeometryBuilder.h:49:10: error: definition of type 'DDSpecPar' conflicts with type alias of the same name [clang-diagnostic-error]
  struct DDSpecPar;
         ^
/cvmfs/cms-ib.cern.ch/nweek-02643/slc7_amd64_gcc820/cms/cmssw/CMSSW_11_2_X_2020-08-27-1100/src/DetectorDescription/DDCMS/interface/DDSpecParRegistry.h:9:9: note: 'DDSpecPar' declared here
--
Geometry/DTGeometryBuilder/plugins/dd4hep/DTGeometryBuilder.h:49:10: error: definition of type 'DDSpecPar' conflicts with type alias of the same name [clang-diagnostic-error]
  struct DDSpecPar;
         ^
/cvmfs/cms-ib.cern.ch/nweek-02643/slc7_amd64_gcc820/cms/cmssw/CMSSW_11_2_X_2020-08-27-1100/src/DetectorDescription/DDCMS/interface/DDFilteredView.h:49:9: note: 'DDSpecPar' declared here
--
gmake: *** [config/SCRAM/GMake/Makefile.coderules:128: code-checks] Error 2
gmake: *** [There are compilation/build errors. Please see the detail log above.] Error 2
```

slomeo commented:
I'll check this error

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-31043/17966

- This PR adds an extra 52KB to repository

Code check has found code style and quality issues which could be resolved by applying following patch(s)
- **code-format**:
https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-31043/17966/code-format.patch
e.g. `curl https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-31043/17966/code-format.patch | patch -p1`
You can also run `scram build code-format` to apply code format directly

slomeo committed:
added lines suggest by @smuzaffar

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-31043/17967

- This PR adds an extra 56KB to repository

smuzaffar commented on Geometry/DTGeometryBuilder/plugins/DTGeometryValidate.cc:
```diff
@@ -82,11 +82,12 @@ class DTGeometryValidate : public one::EDAnalyzer<> {
       outfileName_(iConfig.getUntrackedParameter<string>("outfileName", "validateDTGeometry.root")),
       tolerance_(iConfig.getUntrackedParameter<int>("tolerance", 6)) {
-  fwGeometry_.loadMap(infileName_.c_str());
+  edm::FileInPath fp("Geometry/DTGeometryBuilder/data/cmsRecoGeom-2021.root");
```
@slomeo , may be you can keep the `infileName` as a configurable parameter in validateDTGeometry_cfg.py and use
```
edm::FileInPath fp(infileName_.c_str());
```

cmsbuild commented:
Pull request #31043 was updated. @civanch, @Dr15Jones, @makortel, @cvuosalo, @ianna, @mdhildreth, @cmsbuild can you please check and sign again.

smuzaffar commented on Geometry/DTGeometryBuilder/test/python/validateDTGeometry_cfg.py:
```diff
@@ -37,7 +37,7 @@
 process.valid = cms.EDAnalyzer("DTGeometryValidate",
 
-                               infileName = cms.untracked.string('cmsRecoGeom-2021.root'),
+#                               infileName = cms.untracked.string('cmsRecoGeom-2021.root'),
```
see my comment https://github.com/cms-sw/cmssw/pull/31043/files/2d3b727f5f67fa52bda586c67dc5acbde80f3e77..278777a07be31745322a4ac08abba9488fac5af1#r478965333
its better to keep it configurable. So I would suggest that keep `infileName` and set it to
```
infileName = cms.untracked.string('Geometry/DTGeometryBuilder/data/cmsRecoGeom-2021.root'),
```

slomeo replied:
@smuzaffar : ok, but in validateDTGeometry_cfg.py do I have to write the complete path? infileName = cms.untracked.string('Geometry/DTGeometryBuilder/data/cmsRecoGeom-2021.root') instead infileName = cms.untracked.string('cmsRecoGeom-2021.root') ?

makortel replied:
> So I would suggest that keep `infileName` and set it to
> 
> ```
> infileName = cms.untracked.string('Geometry/DTGeometryBuilder/data/cmsRecoGeom-2021.root'),
> ```

Note that `FileInPath` can be used in python as well, so this could be
```python
infileName = cms.untracked.FileInPath('Geometry/DTGeometryBuilder/data/cmsRecoGeom-2021.root'),
```
and in the C++
```c++
auto fp = iConfig.getUntrackedParameter<edm::FileInPath>("infileName");
```

slomeo commented:
@smuzaffar : I set in validationDTgeometry_cfg.py the line "infileName = cms.untracked.string('Geometry/DTGeometryBuilder/data/cmsRecoGeom-2021.root')" and in DTValidateGeometry.cc I put the line "edm::FileInPath fp(infileName_.c_str());" instead of "edm::FileInPath fp("Geometry/DTGeometryBuilder/data/cmsRecoGeom-2021.root");" but in order to run correctly the validation script I have to add the line "infileName_(iConfig.getUntrackedParameter<string>("infileName", "Geometry/DTGeometryBuilder/data/cmsRecoGeom-2021.root"))," in the constructor before { . In this way compilation and validation are fine. Is it correct?

smuzaffar commented:
yes that is correct. You need to keep the 
```
infileName_(iConfig.getUntrackedParameter("infileName", "Geometry/DTGeometryBuilder/data/cmsRecoGeom-2021.root")),"
```
line. This is needed to read the `infileName` from the parameter set.

slomeo committed:
restored infileName inside the validation script

cmsbuild commented:
The code-checks are being triggered in jenkins.

slomeo commented:
@smuzaffar : done. All unnecessary comments, that I wrote in the code, will be deleted when the test will be ok

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-31043/17970

- This PR adds an extra 60KB to repository

cmsbuild commented:
Pull request #31043 was updated. @civanch, @Dr15Jones, @makortel, @cvuosalo, @ianna, @mdhildreth, @cmsbuild can you please check and sign again.

mrodozov commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.
Test Parameters:
  - **MATRIX_EXTRAS** = 25202.1

Dr15Jones commented on Geometry/DTGeometryBuilder/plugins/DTGeometryValidate.cc:
```diff
@@ -82,11 +82,15 @@ class DTGeometryValidate : public one::EDAnalyzer<> {
-  fwGeometry_.loadMap(infileName_.c_str());
-  outFile_ = new TFile(outfileName_.c_str(), "RECREATE");
+  //edm::FileInPath fp("Geometry/DTGeometryBuilder/data/cmsRecoGeom-2021.root");
+  edm::FileInPath fp(infileName_.c_str());
```
Why is `infileName_` not just a `edm::FileInPath`?

cmsbuild commented:
-1

Tested at: 7a0258422636cd233f8aec4dff61a9c5b0d1b664

CMSSW: CMSSW_11_2_X_2020-08-27-2300
SCRAM_ARCH: slc7_amd64_gcc820
You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c42353/8984/summary.html

I found follow errors while testing this PR

**Failed tests:  UnitTests**

* **Unit Tests**:

I found errors in the following unit tests: 
 
 ---> test GeometryDTGeometryBuilderTestDriver had ERRORS

cmsbuild commented:
Comparison job queued.

slomeo commented:
@cvuosalo : now the problem with the external repo is finally fixed, but the runTest.sh (that fails) contains two different checks: 

1) validateDTGeometry_cfg.py (related to DD4hep, now ok)

2) testDTGeometry.py (if I understood correctly, it is related to DDD; please @ianna tell me if I'm wrong) that is failed as you can see  by the log file. You can find the problem in [1] (i.e No data of type "MuonGeometryConstants" with label "" in record "IdealGeometryRecord")

After that these two different scripts finished  a "diff" command is performed on the log files produced to see if there are differences among printouts (DD4hep vs DDD). 

In order to have DD4hep & DDD validation scripts working well at the same time it is needed what @bsunanda made for Muon Sub systems (for GEM see #30971 )  

IMHO if the comparison test will be ok, I think that it could better to go forward with this PR and in a second time perform an update of the DDD validation script (as @bsunanda always made after my PRs have been merged).

please let me know how I have to proceed 
 
[1] 
===== Test "cmsRun testDTGeometry.py" ====
PersistencyIO    INFO  +++ Set Streamer to dd4hep::OpaqueDataBlock
CompactLoader    INFO  +++ Processing compact file: /cvmfs/cms-ib.cern.ch/nweek-02643/slc7_amd64_gcc820/cms/cmssw/CMSSW_11_2_X_2020-08-27-2300/src/Geometry/CMSCommonData/data/dd4hep/cmsExtendedGeometry2021.xml with flag (null)
DD4CMS           INFO  +++ Processing the CMS detector description file:///cvmfs/cms-ib.cern.ch/nweek-02643/slc7_amd64_gcc820/cms/cmssw/CMSSW_11_2_X_2020-08-27-2300/src/Geometry/CMSCommonData/data/dd4hep/cmsExtendedGeometry2021.xml
Detector         INFO  *********** Created World volume with size: 10100 10100 45000
Detector         INFO  +++ Patching names of anonymous shapes....
DDDefinition     INFO  +++ Finished processing file:///cvmfs/cms-ib.cern.ch/nweek-02643/slc7_amd64_gcc820/cms/cmssw/CMSSW_11_2_X_2020-08-27-2300/src/Geometry/CMSCommonData/data/dd4hep/cmsExtendedGeometry2021.xml
----- Begin Fatal Exception 28-Aug-2020 14:46:35 CEST-----------------------
An exception of category 'NoProxyException' occurred while
   [0] Running EventSetup component DTGeometryESProducer/'
Exception Message:
No data of type "MuonGeometryConstants" with label "" in record "IdealGeometryRecord"
 Please add an ESSource or ESProducer to your job which can deliver this data.
----- End Fatal Exception -------------------------------------------------

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c42353/8984/summary.html

@slava77 comparisons for the following workflows were not done due to missing matrix map:
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-c42353/25202.1_TTbar_13+TTbar_13+DIGIUP15APVSimu_PU25+RECOUP15_PU25+HARVESTUP15_PU25

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 4 differences found in the comparisons
 - DQMHistoTests: Total files compared: 35
 - DQMHistoTests: Total histograms compared: 2609667
 - DQMHistoTests: Total failures: 5
 - DQMHistoTests: Total nulls: 1
 - DQMHistoTests: Total successes: 2609639
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: -0.004 KiB( 34 files compared)
 - DQMHistoSizes: changed ( 10224.0 ): -0.004 KiB       MessageLogger/Warnings
 - Checked 149 log files, 22 edm output root files, 35 DQM output files

cvuosalo commented:
@slomeo `testDTGeometry.py` is a DD4hep test, even though it uses a `fromDDD` variable.  I think you could make the same changes to `testDTGeometry.py` as you did to `validateDTGeometry_cfg.py`, and it should work. Could you please try that?

slomeo commented:
@cvuosalo : I'll try

slomeo committed:
modified testDTGeometry.py

cmsbuild commented:
The code-checks are being triggered in jenkins.

slomeo commented:
@cvuosalo : I modified testDTGeometry.py.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-31043/18024

- This PR adds an extra 68KB to repository

cmsbuild commented:
Pull request #31043 was updated. @civanch, @Dr15Jones, @makortel, @cvuosalo, @ianna, @mdhildreth, @cmsbuild can you please check and sign again.

slomeo commented:
please, someone of the reviewers can perform a "please test"?

mrodozov commented:
allow @slomeo test rights

mrodozov commented:
you try now

slomeo commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.
Test Parameters:
  - **MATRIX_EXTRAS** = 25202.1

slomeo commented:
@mrodozov : tks

cmsbuild commented:
+1
Tested at: 57b6f2f67935396a0079891f4e9c8047153182d1
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c42353/9010/summary.html
CMSSW: CMSSW_11_2_X_2020-08-30-2300
SCRAM_ARCH: slc7_amd64_gcc820

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c42353/9010/summary.html

@slava77 comparisons for the following workflows were not done due to missing matrix map:
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-c42353/25202.1_TTbar_13+TTbar_13+DIGIUP15APVSimu_PU25+RECOUP15_PU25+HARVESTUP15_PU25

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 35
 - DQMHistoTests: Total histograms compared: 2609667
 - DQMHistoTests: Total failures: 1
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2609644
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 34 files compared)
 - Checked 149 log files, 22 edm output root files, 35 DQM output files

slomeo commented:
@cvuosalo : The test and the comparison are ok, now I'll clean the code

slomeo committed:
code cleaned

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-31043/18029

- This PR adds an extra 20KB to repository

cmsbuild commented:
Pull request #31043 was updated. @civanch, @Dr15Jones, @makortel, @cvuosalo, @ianna, @mdhildreth, @cmsbuild can you please check and sign again.

slomeo commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.
Test Parameters:
  - **MATRIX_EXTRAS** = 25202.1

cmsbuild commented:
+1
Tested at: 94fbdcf461a5e66cd53ccf8be180ce38b4d720b6
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c42353/9015/summary.html
CMSSW: CMSSW_11_2_X_2020-08-31-1100
SCRAM_ARCH: slc7_amd64_gcc820

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c42353/9015/summary.html

@slava77 comparisons for the following workflows were not done due to missing matrix map:
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-c42353/25202.1_TTbar_13+TTbar_13+DIGIUP15APVSimu_PU25+RECOUP15_PU25+HARVESTUP15_PU25

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 4 differences found in the comparisons
 - DQMHistoTests: Total files compared: 35
 - DQMHistoTests: Total histograms compared: 2609667
 - DQMHistoTests: Total failures: 5
 - DQMHistoTests: Total nulls: 1
 - DQMHistoTests: Total successes: 2609639
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: -0.004 KiB( 34 files compared)
 - DQMHistoSizes: changed ( 10224.0 ): -0.004 KiB       MessageLogger/Warnings
 - Checked 149 log files, 22 edm output root files, 35 DQM output files

cvuosalo commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @silviodonato, @dpiparo, @qliphy (and backports should be raised in the release meeting by the corresponding L2)

qliphy commented:
+1
